spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -106,5 +127,260 @@ public static Schema getSchemaFromDataFrame(DataFrame rdd) {
+   *
+   * @param schema will use to apply for current DDF
+   * @return Tuple2<SparkDDF,ApplySchemaStatistic> +SparkDDF: new sparkddf which applied new schema already
+   * +ApplySchemaStatistic:  applied schema statistic @see ApplySchemaStatistic
+   */
+  public Tuple2<SparkDDF, ApplySchemaStatistic> applySchema(Schema schema) throws DDFException {
+    long startApplySchema = System.currentTimeMillis();
+    // Keep info to Convert column at <pos> to <ColumnType>
+    List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumns = getListConvertColumn(this.getSchema(), schema);
+
+    try {
+      SparkDDFManager manager = (SparkDDFManager) this.getManager();
+//      SparkContext sparkContext = manager.getSparkContext();
+      Accumulator<ApplySchemaStatistic> statisticAccumulator = manager.getSparkContext().accumulator(new ApplySchemaStatistic(), ApplySchemaStatisticParam.getInstance());
+
+      JavaRDD<Row> rdd = ((SparkDDF) this.getDDF()).getRDD(Row.class).toJavaRDD();
+      JavaRDD<Row> filteredRdd = rdd.filter(row -> convertRowType(row, listConvertColumns, statisticAccumulator));
+      long totalLineSuccess = filteredRdd.count();
+      ApplySchemaStatistic statistic = statisticAccumulator.value();
+      statistic.setTotalLineSuccessed(totalLineSuccess);
+      if(mLog.isDebugEnabled()){
+        mLog.debug("ApplySchema in " + (System.currentTimeMillis() - startApplySchema) + " ms");
+        mLog.debug("Total Line Processed: " + statistic.getTotalLineProcessed());
+        mLog.debug("Total Line Succeed: " + statistic.getTotalLineSuccessed());
+        if(mLog.isTraceEnabled()){
+          statistic.getMapColumnStatistic().forEach( (name,columnStatistic) ->{
+            mLog.trace("column: " + name + " fail in: " + columnStatistic.getNumConvertedFailed());
+            mLog.trace("sample: " + StringUtils.join(columnStatistic.getSampleData(),','));
+          });
+        }
+      }
+      if(statistic.getTotalLineSuccessed() >0) {
+        DataFrame dataFrame = manager.getHiveContext().createDataFrame(filteredRdd.rdd(), filteredRdd.first().schema());
+        return new Tuple2<>((SparkDDF) SparkUtils.df2ddf(dataFrame, manager), statistic);
+      }else{
+        return new Tuple2<>(null, statistic);
+      }
+    } catch (ClassCastException cce) {
+      mLog.error("Exception when down cast from DDFManager to SparkDDFManager", cce);
+      throw new DDFException("applySchema only works with SparkDDFManager");
+    }
+
+  }
+
+  /**
+   * Convert Row Data To Given Type If failed, store data to statistic
+   *
+   * @param row                  which hold multiple columns to convert
+   * @param listConvertColumns   given columnIndex & columnName &  type to convert to
+   * @param statisticAccumulator statistic
+   */
+  private Boolean convertRowType(Row row, List<Tuple3<Integer, String, Schema.ColumnType>> listConvertColumns, Accumulator<ApplySchemaStatistic> statisticAccumulator) {
```

Comment:
this function does not do any conversion yet, it's just checking whether you can convert a row or not.